### DISCUSSION TITLE
File manager integration

### DISCUSSION THREAD

**Discussion by Thatoo** [AppImage/AppImageKit#1364] [Ideas]
Nextcloud's appimage is great but I came back using the debian package because it doesn't integrate into nautilus.

There is a long thread talking about this issue in nextcloud repo, https://github.com/nextcloud/desktop/issues/2217, and integration between the appimage and file managers (nautilus|dolphin|caja|nemo) seems quiet difficult.

Is there any hint to share with the nextcloud team to solve this issue?

**Comment by probonopd** [comment]
Nextcloud would need to add code that would write the necessary files to the respective locations within `$HOME` (e.g., upon each launch of the Nextcloud client). When writing such code, they could check for the presence of the `$APPIMAGE` environment variable and use its value if available. 
When doing this, they might also want to add an option to uninstall such desktop integration (by deleting the respective files). 
I'd be happy to support Nextcloud devs if questions arise.

I don't know whether ownCloud has more functioniality regarding this, but it probably wouldn't hurt to check.
